crypto/digestible/derive/src/lib.rs

```diff
@@ -545,6 +547,101 @@ fn try_digestible_enum(
     Ok(expanded.into())
 }
 
+fn try_digestible_enum_transparent(
+    ident: &Ident,
+    generics: &Generics,
+    variant_data: &DataEnum,
+) -> Result<TokenStream, &'static str> {
```

Comment:
I think there maybe should be some extra checks here around things like all variants must be non-unit and contain distinct, non-transparent, inner types.

Reply:
That's an interesting idea

I'm not sure that it is technically possible to check if types are transparent in a proc macro. Proc macros can only see the code for item that the proc macro is tagging directly, it can't go look at the code for another structure.

We might be able to do something where we check if the token streams for the types are all different.
Or, we could like, add a `pub const _IGNORED: _ = { match .. }` to the code gen which will fail to compile if the types are not all different because the match statement would be ambiguous or something

Reply:
Okay we have now banned unit variants, and we check if the inner types are the same